x/stakeibc/keeper/msg_server_submit_tx.go

```diff
@@ -170,8 +171,41 @@ func (k Keeper) UpdateWithdrawalBalance(ctx sdk.Context, zoneInfo types.HostZone
 	)
 }
 
+func (k Keeper) SubmitTxsDayEpoch(ctx sdk.Context, connectionId string, msgs []sdk.Msg, account types.ICAAccount) (uint64, error) {
+	sequence, err := k.SubmitTxsEpoch(ctx, connectionId, msgs, account, epochstypes.DAY_EPOCH)
+	if err != nil {
+		return 0, err
+	}
+	return sequence, nil
+}
+
+func (k Keeper) SubmitTxsStrideEpoch(ctx sdk.Context, connectionId string, msgs []sdk.Msg, account types.ICAAccount) (uint64, error) {
+	sequence, err := k.SubmitTxsEpoch(ctx, connectionId, msgs, account, epochstypes.STRIDE_EPOCH)
+	if err != nil {
+		return 0, err
+	}
+	return sequence, nil
+}
+
+func (k Keeper) SubmitTxsEpoch(ctx sdk.Context, connectionId string, msgs []sdk.Msg, account types.ICAAccount, epochType string) (uint64, error) {
+	epochInfo, found := k.epochsKeeper.GetEpochInfo(ctx, epochType)
+	if !found {
+		k.Logger(ctx).Error("Failed to get epoch info for %s", epochType)
+		return 0, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Failed to get epoch info for %s", epochType)
+	}
+	// BUFFER by 5% of the epoch length
+	BUFFER := epochInfo.GetDuration().Nanoseconds() / 20
```

Comment:
should we make the buffer (e.g. 20) a param?

Reply:
Done.